CHANGELOG.md

```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+#### Bugfixes
+- Make the additional lifecycle events match the docs
```

Comment:
It's a bit unclear what was exactly changed, probably should describe the behavior that could be observed (like endless loop due to use of lifecycle events).

Reply:
I think this is sufficient because:
1) there are, to my knowledge, no users of this API, so no-one will have run into this
2) The pr is readily available
3) I don't want to make the changelog be really long with useless content

But I'll put whatever you prefer, if you give me the text you want to put here

Reply:
I mean, `- Fix infinite loop when using registering a source with lifecycle events` is short enough from what I can say? Or is it not reflecting the situation correctly?

Reply:
So the infinite loop was only because the events were never getting read - we were still making progress, and other sources would have worked.

That is, the infinite loop is in the WaylandSource - yes because of this issue, but that isn't the only way this could manifest

Reply:
Hm, yeah, that's actually fair. I'm fine with the changelog like that, it's just reads without providing a real value to the user.

Reply:
probably  @elinorbgr has a better suggestion on that matter, since we'd need to patch release and maybe yank 0.12.

Reply:
Overall, I'd prefer to have a quick description of what the bug was, so something like this would be fine imo:

```
- Fix `EventSource::before_handle_events()` being erroneously give an iterator over synthetic events instead of real events
```